### PR TITLE
NRPT-657:Agri role updates.

### PR DESCRIPTION
**Pull request by marklise**

### CODE REVIEW

**Comment by marklise** [comment]
Create admin:agri record:

```POST http://localhost:3000/api/record
{
    "orders": [
        {
            "recordName": "AGRI Role Test",
            "recordSubtype": "None",
            "issuingAgency": "Ministry of Agriculture",
            "author": "BC Government",
            "issuedTo": {
                "type": "Individual",
                "companyName": null,
                "firstName": "Test",
                "middleName": "Middle",
                "lastName": "Name",
                "fullName": null,
                "dateOfBirth": null
            },
            "OrderNRCED": {
                "summary": "NRCED Summary",
                "addRole": "public"
            },
            "OrderLNG": {
                "description": "LNG Summary",
                "addRole": "public"
            }
        }
    ]
}
```
Verify success and record read/write issuedTo read/write have admin:agri role

Create a document in the record:

```POST http://localhost:3000/api/record/{record_id}/document
form-data
fileName: test-file-name.pdf
upfile: browse to your file
```
Verify success document record has admin:agri role

To verify redaction, you can create a record in NRPTI from another user role and set the individual age to under 19. Then do a search using:

```
GET http://localhost:3000/api/search?dataset=Order,Inspection,Certificate,Permit,SelfReport,Agreement,RestorativeJustice,Ticket,AdministrativePenalty,AdministrativeSanction,Warning,ConstructionPlan,ManagementPlan,CourtConviction,AnnualReport,CertificateAmendment,Correspondence,DamSafetyInspection,Report&pageNum=0&pageSize=25&sortBy=-dateAdded&fields=
```

File: api/src/tasks/import-task.js [hunk 1]
```diff
@@ -173,6 +173,10 @@ async function runTask(nrptiDataSource, auth_payload, params = null, recordTypes
     additionalReadRoles.push(ApplicationRoles.ADMIN_FLNRO)
   }
 
+  if(nrptiDataSource.dataSourceLabel == 'mis-csv' && params == 'Inspection'){
+    additionalReadRoles.push(ApplicationRoles.ADMIN_AGRI)
```
Comment (BcGovNeal) [inline api/src/tasks/import-task.js:177]: Missing the AGRI CMDB source here as well

**Review by mtCarto — COMMENTED** [review]
Records and documents have the correct roles when created, redaction is functioning as expected. Addressing Neal's comments should complete the UI portion of this role as well.

File: api/src/tasks/import-task.js (hunk 1 above)
Comment (marklise) [reply api/src/tasks/import-task.js:181]: Fixed in a41b100

File: angular/projects/admin-nrpti/src/app/import/import-csv/import-csv.component.ts [hunk 2]
```diff
@@ -52,6 +52,14 @@ export class ImportCSVComponent implements OnInit {
       this.csvTypes = {
         'coors-csv': ['Administrative Sanction']
       };
+    } else if (this.factoryService.userOnlyInLimitedRole(Constants.ApplicationRoles.ADMIN_AGRI)) {
+      this.dataSourceTypes = [
+        { displayName: 'AGRI-MIS', value: 'mis-csv' }
```
[the remark this answers is not in the document]
Comment (marklise) [reply angular/projects/admin-nrpti/src/app/import/import-csv/import-csv.component.ts:58]: Fixed in a41b100

**Review by mtCarto — APPROVED** [review]

**Review by BcGovNeal — COMMENTED** [review]
Looks good!

**Review by BcGovNeal — APPROVED** [review]

**Pull request merged**